### PR TITLE
Task 7 create ts model

### PR DESCRIPTION
**Pull request by Chips421**
Created an initial time series model using Holt-Winters forecasting in the file Preliminary_TS_Model.ipynb.

Renamed Almanac/get_weather_data.py to Almanac/Data.py and updated import statements where needed.

Created the Binarize function to binarize weather data and added it to Almanac/Data.py

### CODE REVIEW

File: Almanac/Data.py [hunk 1]
```diff
@@ -1,85 +1,131 @@
...
+    data = ms.Daily(location, start, end)
+    data = data.fetch()
+    return data
+
+
+
+
+def binarize(df,cols):
+    '''
+    A function to binarize weather data. Sets values less than 0 to 1 and sets values greater then 0 to 0.
```
Comment (awdleishman) [inline Almanac/Data.py]: Can we make the threshold a parameter of this function, with a default value of zero?

File: Almanac/Data.py [hunk 2]
```diff
@@ -1,85 +1,131 @@
...
+        
+        
+    Returns:
+    
+    df : pandas.DataFrame
+        A DataFrame containing the original columns of df and the new columns of binarized data.
+    '''
+    
+    
+    if type(cols) == str:
```
Comment (awdleishman) [inline Almanac/Data.py]: I think these should use isinstance()

Per flake8:

`do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`Flake8(E721)`

File: Almanac/Data.py [hunk 3]
```diff
@@ -1,85 +1,131 @@
...
+    location = ms.Point(lat, lon)
+
+    data = ms.Daily(location, start, end)
+    data = data.fetch()
+    return data
+
+
+
+
+def binarize(df,cols):
```
Comment (awdleishman) [inline Almanac/Data.py]: There's some formatting here that black & flake8 disagree with

File: .ipynb_checkpoints/Binarize-checkpoint.ipynb
Comment (awdleishman) [inline .ipynb_checkpoints/Binarize-checkpoint.ipynb:1]: What's the difference between the checkpoints and the notebooks?

File: Almanac/__pycache__/Data.cpython-311.pyc
Comment (awdleishman) [inline Almanac/__pycache__/Data.cpython-311.pyc:1]: We should add these cache-type files to the .gitignore

**Review by awdleishman — COMMENTED** [review]
We should be importing binarize, not redefining it in the notebook

**Review by awdleishman — CHANGES_REQUESTED** [review]
Please fix the formatting/ linting issues in Data.py and remove the cache files

File: Almanac/Data.py (hunk 1 above)
Comment (Chips421) [reply Almanac/Data.py]: Yes.

File: Almanac/Data.py (hunk 2 above)
Comment (Chips421) [reply Almanac/Data.py]: Done.

File: Almanac/Data.py (hunk 3 above)
Comment (Chips421) [reply Almanac/Data.py]: Resolved.

File: .ipynb_checkpoints/Binarize-checkpoint.ipynb
Comment (Chips421) [reply .ipynb_checkpoints/Binarize-checkpoint.ipynb:1]: TBH. I didn't even notice that the checkpoints folder had been created, but they are essentially the same I think. I've removed the checkpoints from the directory for the latest commit.

File: Almanac/__pycache__/Data.cpython-311.pyc
Comment (Chips421) [reply Almanac/__pycache__/Data.cpython-311.pyc:1]: Sounds good. Is that something I need to do on my end or is that a repository thing?

File: Almanac/Data.py [hunk 4]
```diff
@@ -1,85 +1,129 @@
...
+        end = datetime.strptime(end, "%Y-%m-%d")
+
+    location = ms.Point(lat, lon)
+
+    data = ms.Daily(location, start, end)
+    data = data.fetch()
+    return data
+
+
+def binarize(df, cols, thresh=0):
```
Comment (awdleishman) [inline Almanac/Data.py:88]: Add `thresh` to docstring

File: Almanac/__pycache__/Data.cpython-311.pyc
Comment (awdleishman) [reply Almanac/__pycache__/Data.cpython-311.pyc:1]: Thank you for removing the cache files. You can add the gitignore as part of this PR if you'd like

The gitignore will keep that folder from being tracked in the future.

Here's some docs:

https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files

https://www.freecodecamp.org/news/gitignore-file-how-to-ignore-files-and-folders-in-git/

**Review by awdleishman — CHANGES_REQUESTED** [review]
Please update the docstring

**Review by awdleishman — APPROVED** [review]

**Comment by awdleishman** [comment]
[Preliminary TS model building](https://trello.com/c/os4WM90J/7-preliminary-ts-model-building)

**Pull request merged**